lib/grammar.js

```diff
@@ -0,0 +1,328 @@
+
+// https://www.w3.org/TR/xml11/#NT-CharData
+// [14] CharData    ::= [^<&]* - ([^<&]* ']]>' [^<&]*)
+
+// https://www.w3.org/TR/xml11/#NT-Comment
+// [15] Comment    ::= '<!--' ((Char - '-') | ('-' (Char - '-')))* '-->'
+var Comment = reg('<!--', regg(chars_without(Char, '-'), '|', regg('-', chars_without(Char, '-'))), '*', '-->');
+
+// https://www.w3.org/TR/xml/#NT-Mixed
+// [51] Mixed ::= '(' S? '#PCDATA' (S? '|' S? Name)* S? ')*' | '(' S? '#PCDATA' S? ')'
+// [VC: Proper Group/PE Nesting] [VC: No Duplicate Types]
+var Mixed = regg(
+	reg(/\(/, S_OPT, '#PCDATA', regg(S_OPT, /\|/, S_OPT, Name), '*', S_OPT, /\)\*/),
+	'|',
+	reg(/\(/, S_OPT, '#PCDATA', S_OPT, /\)/)
+);
+
+var _children_quantity = /[?*+]?/;
+// [49] choice ::= '(' S? cp ( S? '|' S? cp )+ S? ')' [VC: Proper Group/PE Nesting]
+// [50] seq ::= '(' S? cp ( S? ',' S? cp )* S? ')' [VC: Proper Group/PE Nesting]
+// simplification to solve circular referencing, but doesn't check validity constraint "Proper Group/PE Nesting"
+var _choice_or_seq = reg('[', NameChar.chars, S.chars, chars(_children_quantity), '()|,]*');
+// [48] cp ::= (Name | choice | seq) ('?' | '*' | '+')?
+//               === (Name | '(' S? cp ( S? '|' S? cp )+ S? ')' | '(' S? cp ( S? ',' S? cp )* S? ')') ('?' | '*' | '+')?
+//               !== (Name | [_choice_or_seq]*) ('?' | '*' | '+')?
+// simplification to solve circular referencing, but doesn't check validity constraint "Proper Group/PE Nesting"
+var cp = reg(regg(Name, '|', _choice_or_seq), _children_quantity);
+var choice = regg(/\(/, S_OPT, cp, regg(S_OPT, /\|/, S_OPT, cp), '+', S_OPT, /\)/);
+var seq = regg(/\(/, S_OPT, cp, regg(S_OPT, /,/, S_OPT, cp), '*', S_OPT, /\)/);
+
+// [47] children ::= (choice | seq) ('?' | '*' | '+')?
+// simplification to solve circular referencing, but doesn't check validity constraint "Proper Group/PE Nesting"
+var children = reg(regg(choice, '|', seq), _children_quantity);
+
+// https://www.w3.org/TR/xml11/#NT-contentspec
+// [46] contentspec ::= 'EMPTY' | 'ANY' | Mixed | children
+var contentspec = regg('EMPTY', '|', 'ANY', '|', Mixed, '|', children);
+
+// https://www.w3.org/TR/xml11/#NT-elementdecl
+// [45] elementdecl ::= '<!ELEMENT' S Name S contentspec S? '>'
+// https://www.w3.org/TR/xml-names/#NT-elementdecl
+// [17] elementdecl ::= '<!ELEMENT' S QName S contentspec S? '>'
+// because of https://www.w3.org/TR/xml11/#NT-PEReference
+// they also need to support PEReference when checked before replacement of entities declared in DTD (which xmldom is not supporting)
+var elementdecl = reg('<!ELEMENT', S, regg(QName, '|', PEReference), S, regg(contentspec, '|', PEReference), S_OPT, '>');
+
+// [58] NotationType ::= 'NOTATION' S '(' S? Name (S? '|' S? Name)* S? ')'
+//     [VC: Notation Attributes] [VC: One Notation Per Element Type] [VC: No Notation on Empty Element] [VC: No Duplicate Tokens]
+var NotationType = reg('NOTATION', S, /\(/, S_OPT, Name, regg(S_OPT, '|', S_OPT, Name), '*', S_OPT, /\)/);
```

Comment:
```suggestion
var NotationType = reg('NOTATION', S, /\(/, S_OPT, Name, regg(S_OPT, /\|/, S_OPT, Name), '*', S_OPT, /\)/);
```

The pipe should be a literal pipe.